middle_end/flambda2/tests/mlexamples/tests3.flt

```diff
@@ -126,15 +126,18 @@ let code loopify(never) size(65) newer_version_of(foo_0)
     where k2 (apply_result) =
       ((let cond_result = %is_flat_float_array arr in
         switch cond_result
-          | 0 -> k3
-          | 1 -> k2)
-         where k3 =
+          | 0 -> k4
+          | 1 -> k3)
+         where k4 =
            let ifnot_result = %array_set arr.(i) <- apply_result in
-           cont k (0)
-         where k2 =
+           cont k2
+         where k3 =
            let prim = %unbox_float apply_result in
            let ifso_result = %array_set float arr.(i) <- prim in
-           cont k (0))
+           cont k2
+         where k2 =
+           let if_then_else_result = 0 in
+           cont k (if_then_else_result))
```

Comment:
It does not change much, but is that expected ?

Reply:
From what I recall, it is expected, because `0` can now be discovered as a dominator, thus we insert the let_binding at the start of the continuation.

Reply:
but you might need to look at the rawflambda to really see/understand what is happening